org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/processing/JobManager.java

```diff
@@ -548,6 +549,13 @@ protected IStatus run(IProgressMonitor monitor) {
 			throw e;
 		}
 	}
+
+	/**
+	 * Notified when the when queue is found to be empty by the indexer loop.
+	 */
+	protected void onJobQueueEmpty() {
```

Comment:
can use "abstract"?

Reply:
The reason I didn't made it abstract because this method is an optional method which a subclass must implement only if it wants to get notified on job queue is empty. But since there is only one subclass we could make is abstract as well. I'm ok with either, let me know your thought on it.

Reply:
Since this (empty) implementation is never called it is useless. It is not important though.